test/util/etcd.go

```diff
@@ -132,12 +131,44 @@ func (s EtcdTestServer) DumpEtcdOnFailure(t *testing.T) {
 		t.Logf("error dumping etcd: %v", err)
 		return
 	}
-	jsonResponse, err := json.Marshal(response.Node)
+
+	writeEtcdDump(t, response.Node)
+}
+
+func DumpEtcd3(t *testing.T) {
```

Comment:
Would be better to take an etcd client here rather than implicitly creating one.